src/signature.sh

```diff
@@ -0,0 +1,348 @@
+          options_values['ERROR']='The option --add-fixes or -f demands an argument'
+          return 22 # EINVAL
+        fi
+
+        options_values['ADD_FIXES']=1
+        options_values['FIXES_LINE']="$(format_fixes_message "$(str_strip "$2")")"
+
+        if [[ "$?" -gt 0 ]]; then
+          options_values['ERROR']="Invalid commit reference with --add-fixes or -f: ${2}"
+          return 22 # EINVAL
+        fi
+
+        options_values['FIXES_LINE']="${TRAILER_TAGS['FIXES']} ${options_values['FIXES_LINE']}"
+        trailer_operations+=('ADD_FIXES')
+        shift 2
+        ;;
+      --verbose)
+        options_values['VERBOSE']=1
+        shift
+        ;;
+      --)
+        # End of options, beginning of arguments.
+        # Overwrite default value if at least one argument is given.
+        [[ -n "$2" ]] && options_values['PATCH_OR_SHA']=''
+        shift
+        ;;
+      *)
+        # Get all passed arguments each loop
+        if [[ "$1" == *"*"* ]]; then
+          # Expand the glob and loop through each resulting file
+          for arg in $1; do
+            options_values['PATCH_OR_SHA']+=" $arg"
+          done
+        else
+          options_values['PATCH_OR_SHA']+=" $1"
+        fi
+        shift
+        ;;
+    esac
+  done
+}
+
+function signature_help()
+{
+  if [[ "$1" == --help ]]; then
+    include "${KW_LIB_DIR}/help.sh"
+    kworkflow_man 'handle-trailer'
+    return
+  fi
+  printf '%s\n' 'kw handle-trailer:' \
+    '  handle-trailer (--add-signed-off-by | -s) (<name>) [<patchset> | <sha>] - Add Signed-off-by' \
+    '  handle-trailer (--add-reviewed-by | -r) (<name>) [<patchset> | <sha>] - Add Reviewed-by' \
+    '  handle-trailer (--add-acked-by | -a) (<name>) [<patchset> | <sha>] - Add Acked-by' \
+    '  handle-trailer (--add-fixes | -f) [<fixed-sha>] [<patchset> | <sha>] - Add Fixes' \
+    '  handle-trailer (--verbose) - Show a detailed output'
+}
+
+# This function registers a signature operation into
+# trailer_operations that will be properly excuted later.
+# It attempts to use the user's name and email configured
+# if a signature is not passed as argument and gives
+# an error if they are not properly set with git config.
+#
+# @keyword Holds a string that is a keyword for a specific operation.
+# @signature Holds a string like 'NAME <EMAIL>' defining the signature.
+function register_signature_operation()
+{
+  local keyword="$1"
+  local signature="$2"
+
+  local operation_key="ADD_${keyword}"
+  local signature_key="${keyword}_LINE"
+  local formated_output
+
+  options_values["$operation_key"]=1
+  options_values["$signature_key"]="${TRAILER_TAGS[$keyword]} "
+
+  # Use default from git config if no argument was given
+  if [[ ! "$signature" ]]; then
+    formated_output="$(format_name_email_from_user)"
+    if [[ "$?" -gt 0 ]]; then
+      return 1
+    fi
+    options_values["$signature_key"]+="$formated_output"
+  else
+    options_values["$signature_key"]+="$(str_strip "$signature")"
```

Comment:
You can't trust the input; you should have some level of validation, such as checking for the string size. You can also check if the email has some minimum stuff such as one @ and at least one .; I don't think you need to implement this, iirc, we already have an email validation code.

Reply:
We can take this route but add a TODO comment to refactor this later. Ideally, we want to extract that function from a generic place like `kw_string.sh`.